core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -60,6 +60,51 @@ private[spark] object CallSite {
   val empty = CallSite("", "")
 }
 
+/** An utility class to map short compression codec names to qualified ones. */
+private[spark] class ShortCompressionCodecNameMapper {
```

Comment:
I also agree - this abstract class is too much. I think just having lz4/bzip2 etc in different places isn't that big of a deal.